### PR TITLE
update node12 to node16 in `action.yml`

### LINKED ISSUE #2196: Github Action Deprecation notices
### Issue workflow progress

_Progress of the issue based on the
[Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [ ] 1. The issue provides a reproduction available on GitHub, Stackblitz or CodeSandbox
     > Make sure to fork this template and run `yarn generate` in the terminal.
     >
     > Please make sure the Codegen and plugins version under `package.json` matches yours.
- [ ] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [ ] 4. A pull request is pending review

---

**Describe the bug**

Github Actions are reporting some impending deprecations and changes that could affect functionality.

```
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: kamilkisiela/graphql-inspector
```

```
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

**To Reproduce** Steps to reproduce the behavior:

Add the action to a workflow
```yml
name: Check GraphQL Schema

on:
  pull_request:
    branches:
      - main

jobs:
  check-graphql-schema:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@master
        
      - uses: kamilkisiela/graphql-inspector@v3.4.0
        with:
          schema: 'main:schema.graphql'

```

**Expected behavior**

Deprecations don't show

**Environment:**

- OS: `ubuntu-latest`
- `@graphql-inspector/...`: 3.4.0
- `graphql`: -
- NodeJS: 12

**Additional context**

### PR DESCRIPTION
**Pull request by booklearner**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

_Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request._

## Description

This action is already using Node.js v16, but the `action.yml` file still references v12. This causes a deprecation notice in GitHub Actions:

```
Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: kamilkisiela/graphql-inspector@master. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.
```

Fixes #2196 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots/Sandbox (if appropriate/relevant):

Adding links to sandbox or providing screenshots can help us understand more about this PR and take action on it as
appropriate

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Environment**:

- OS:
- `@graphql-inspector/...`:
- NodeJS:

## Checklist:

- [ ] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the
      style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you
did and what alternatives you considered, etc...

### CODE REVIEW

**Review by saihaj — APPROVED** [review]

**Review by wokkaflokka — COMMENTED** [review]
Thanks @booklearner @saihaj!

**Comment by azeemchauhan** [comment]
@booklearner  I don't have exposure to this code base, but changing Node12-Node16 also would be required to change the internals where it is calling the @action/core method [setOutput](https://github.com/kamilkisiela/graphql-inspector/blob/096863f00dc60449ac0b433dee61372a4b17dc28/packages/action/src/run.ts#L204) in order to remove the deprecation warning, here is the quick workaround - https://github.com/actions/toolkit/issues/1218#issuecomment-1288890856

**Comment by booklearner** [comment]
Thanks for the note @azeemchauhan, I believe the `setOutput` deprecation notice is unrelated to the Node notice. This codebase can/should move away from `setOutput` by [May 2023](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) but that can be in a separate PR.

cc @kamilkisiela

File: action.yml [hunk 1]
```diff
@@ -44,5 +44,5 @@ outputs:
   changes:
     description: Total number of changes
 runs:
-  using: node12
-  main: action/index.js
+  using: 'node16'
+  main: 'action/index.js'
```
Comment (nathanchapman) [inline action.yml:48]: @booklearner this should fix the test failures
```suggestion
  using: node16
  main: action/index.js
```

Comment (booklearner) [reply action.yml]: Thanks!

**Comment by nathanchapman** [comment]
@kamilkisiela can we please get this merged in?

**Review by TuvalSimha — APPROVED** [review]

**Pull request merged**